### PR TITLE
Fixes #82 - Support for custom HTTP headers

### LINKED ISSUE #82: Support Authorization: header for connection to ElasticSearch
First of all, thanks for sharing your code.

I would like to use it to perform queries against the log store of RedHat's Openshift, but Openshift places its ElasticSearch cluster behind an oAuth proxy that requires Bearer Authentication (please see [this](https://docs.openshift.com/container-platform/4.5/logging/config/cluster-logging-log-store.html#cluster-logging-elasticsearch-exposing_cluster-logging-store))

The Elasticsearch constructor supports an [additional parameter `headers`](https://elasticsearch-py.readthedocs.io/en/master/connection.html#elasticsearch.Urllib3HttpConnection), where we could add an `Authorization: Bearer ....` header. That would probably mean adding a new cli flag to provide the bearer token.

I'll try to submit a PR for this, if you agree.

### PR DESCRIPTION
**Pull request by rg2011**
Adds a CLI flag (-H, --header) to support custom HTTP headers in requests to ElasticSearch, as discussed in issue #82

### CODE REVIEW

File: prometheus_es_exporter/__init__.py
Comment (braedon) [inline prometheus_es_exporter/__init__.py:526]: Not essential, but this parsing/validation could be done in a [callback on the click option](https://click.palletsprojects.com/en/7.x/options/#callbacks-for-validation).

There's an [example in the similar prometheus-mysql-exporter project](https://github.com/braedon/prometheus-mysql-exporter/blob/29b2483b4f0153452e9e433ac395d333584f21b7/prometheus_mysql_exporter/__init__.py#L106-L116) if you want to make the switch.

**Review by braedon — COMMENTED** [review]
Looks good, thanks @rg2011.

Left one suggestion for an alternative way to do the parameter validation and parsing, but I'll leave it up to you whether you want to convert the code or not - what you've got is fine.

Other than that, once the merge conflict is sorted, we'll be good to go.

File: prometheus_es_exporter/__init__.py
Comment (rg2011) [reply prometheus_es_exporter/__init__.py]: Makes sense, I changed that in 22cd35f . Incidentally, it also solved the merge conflict.

**Review by braedon — APPROVED** [review]

**Comment by braedon** [comment]
Released in [0.13.0](https://github.com/braedon/prometheus-es-exporter/releases/tag/0.13.0)

**Pull request merged**
